### PR TITLE
Add tests to test urlencoded IAM policies

### PR DESCRIPTION
**Pull request by dfangl**
## Motivation
During investigation of an IAM issue, I noticed our IAM implementation not returning exactly the same policies as submitted.

This happens, when parts of the policy are already urlencoded, and the response removes this encoding, returning the actual value.
This behavior however does not match AWS, which does return the policies as they were submitted.

A fix for this behavior is in getmoto/moto#8157, this PR only contains AWS validated tests for this behavior. The tests will fail, as long as the PR in moto is not merged and moto updated in LocalStack.

## Changes
* Add AWS validated tests to verify the policies being unchanged, even if they contain urlencoded parts

### CODE REVIEW

**Review by dominikschubert — APPROVED** [review]
LGTM! Tests should also now pass here after a rebase since we've upgraded our moto version since then :+1:

**Comment by dfangl** [comment]
We need to wait for #11710 before rebasing and merging :+1:

**Review by pinzon — APPROVED** [review]
The PR you were waiting for is merged. Maybe is time to merge this one too :+1:

**Pull request merged**
